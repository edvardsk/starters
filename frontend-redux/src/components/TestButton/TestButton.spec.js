import React from 'react';
import Button from './TestButton';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import TestComponent from 'helpers/testHelper';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <TestComponent>
        <Button type='button'>Button</Button>
      </TestComponent>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Button receive type and class', () => {
  const button = shallow(
    <TestComponent>
      <Button className='test' type='button'>Button</Button>
    </TestComponent>
  );
  expect(button.children().props().type).toEqual('button');
  expect(button.children().props().className).toEqual('test');
});
