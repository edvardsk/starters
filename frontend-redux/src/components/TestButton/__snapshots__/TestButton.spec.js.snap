// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 TestButton-button-0-2-1"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-2"
  >
    Button
  </span>
  <span
    className="MuiTouchRipple-root-30"
  />
</button>
`;
